Fix OpenSBI build with GCC 14

GCC 14 is stricter about:
1. Uninitialized variable 'cnt' in fdt_reset_thead.c
2. fence.i instruction requires zifencei extension

--- a/lib/utils/reset/fdt_reset_thead.c
+++ b/lib/utils/reset/fdt_reset_thead.c
@@ -59,7 +59,7 @@ static int thead_reset_init(void *fdt, int nodeoff,
 {
 	void *p;
 	const fdt64_t *val;
 	const fdt32_t *val_w;
-	int len, i, cnt;
+	int len, i, cnt = 0;
 	u32 t, tmp = 0;

 	/* Prepare clone csrs */
--- a/lib/sbi/sbi_tlb.c
+++ b/lib/sbi/sbi_tlb.c
@@ -173,7 +173,7 @@ void sbi_tlb_local_hfence_gvma(struct sbi_tlb_info *tinfo)

 void sbi_tlb_local_fence_i(struct sbi_tlb_info *tinfo)
 {
-	__asm__ __volatile("fence.i");
+	__asm__ __volatile__(".insn 0x100f");
 }

 static void sbi_tlb_entry_process(struct sbi_tlb_info *tinfo)
